File: src/frontend/src/types/order.ts
// Types for Order
export type OrderStatus = 'submitted' | 'confirmed' | 'preparing' | 'ready' | 'completed' | 'cancelled' | 'rejected'
export type PaymentStatus = 'pending' | 'paid' | 'failed' | 'refunded' | 'cancelled'

export interface OrderItem {
  id: string
  order_id: string
  item_id: string
  item_name_zh: string
  item_name_en: string
  quantity: number
  unit_price: number
  options: Record<string, unknown>
  notes?: string
  print_group: string
}

export interface Order {
  id: string
  seat_id: string
  status: OrderStatus
  payment_status: PaymentStatus
  payment_method?: string
  payment_intent_id?: string
  subtotal_sgd: number
  tax_sgd: number
  total_sgd: number
  notes?: string
  customer_notes?: string
  rejected_reason?: string
  items: OrderItem[]
  created_at: string
  updated_at: string
  paid_at?: string
  completed_at?: string
  cancelled_at?: string
}

export interface CreateOrderRequest {
  seat_id: string
  items: {
    item_id: string
    quantity: number
    options: Record<string, unknown>
    notes?: string
  }[]
  customer_notes?: string
}

File: src/backend/app/schemas/order.py
# backend/app/schemas/order.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, Field


class OrderItemCreate(BaseModel):
    item_id: str
    quantity: int = Field(ge=1)
    options: dict = Field(default_factory=dict)
    notes: Optional[str] = None


class OrderCreate(BaseModel):
    seat_id: str
    items: List[OrderItemCreate]
    customer_notes: Optional[str] = None


class OrderItemResponse(BaseModel):
    id: str
    item_id: str
    item_name_zh: str
    item_name_en: str
    quantity: int
    unit_price: float
    options: dict
    notes: Optional[str] = None
    print_group: str

    class Config:
        from_attributes = True


class OrderResponse(BaseModel):
    id: str
    seat_id: str
    status: str
    payment_status: str
    payment_method: Optional[str] = None
    payment_intent_id: Optional[str] = None
    items: List[OrderItemResponse] = []
    subtotal_sgd: float
    tax_sgd: float
    total_sgd: float
    notes: Optional[str] = None
    customer_notes: Optional[str] = None
    rejected_reason: Optional[str] = None
    created_at: datetime
    updated_at: datetime
    paid_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    cancelled_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class OrderStatusResponse(BaseModel):
    id: str
    status: str
    payment_status: str
    updated_at: datetime


class OrderStatusUpdate(BaseModel):
    status: str


class OrderRejectRequest(BaseModel):
    reason: str
    items: Optional[List[str]] = None  # UUIDs of order_items; if None, reject all


class OrderCancelRequest(BaseModel):
    reason: Optional[str] = None
